File: packages/dotnet/src/SlyBrowser/OfficialTrust.cs
namespace SlyBrowser;

internal static class OfficialTrust
{
    private static byte[] Decode(string value) => Convert.FromBase64String(
        value.Replace('-', '+').Replace('_', '/') + new string('=', (4 - value.Length % 4) % 4));

    internal static LicenseServiceClientOptions Create() => new()
    {
        LicenseTrustedKeys = new Dictionary<string, byte[]>(StringComparer.Ordinal)
        {
            ["launch-candidate-20260824"] = Decode("h9ie2nXxsXVKlxjFIz-or1otChHTF8HS94vV_EjY1KM"),
        },
        ReleaseTrustedKeys = new Dictionary<string, byte[]>(StringComparer.Ordinal)
        {
            ["release-launch-candidate-20260824"] = Decode("SvSlPQKT9oZ4nIVuJXgd2pFOC0QblDph29vKlz6NJZo"),
        },
        LicenseFileTrustedKeys = new Dictionary<string, byte[]>(StringComparer.Ordinal)
        {
            ["license-file-private-preview-v1"] = Decode("wc3DR5wOqazjZF_3n41EF1cMh5d-qGv2wkZHyd0Sj6s"),
        },
    };
}

File: packages/python/src/slybrowser/_official_trust.py
"""Bundled public verification keys. This module is not part of the user API."""

from __future__ import annotations

import base64


def _decode(value: str) -> bytes:
    return base64.urlsafe_b64decode(value + "=" * (-len(value) % 4))


LICENSE_LEASE_KEYS = {
    "launch-candidate-20260824": _decode("h9ie2nXxsXVKlxjFIz-or1otChHTF8HS94vV_EjY1KM"),
}
RELEASE_MANIFEST_KEYS = {
    "release-launch-candidate-20260824": _decode("SvSlPQKT9oZ4nIVuJXgd2pFOC0QblDph29vKlz6NJZo"),
}
LICENSE_FILE_KEYS = {
    "license-file-private-preview-v1": _decode("wc3DR5wOqazjZF_3n41EF1cMh5d-qGv2wkZHyd0Sj6s"),
}
